docs/installation/production/additional/official_docker.rst

```diff
@@ -0,0 +1,423 @@
+
+
+    #########################
+    # CONFIGURE ENVIRONMENT #
+    #########################
+    EXPOSE 80
+
+
+    ################
+    # COPY IN SALT #
+    ################
+    ADD docker/salt/ /srv/salt/
+
+
+    #######
+    # RUN #
+    #######
+    CMD bash run.sh
+
+Run.sh
+------
+
+The bash script ``run.sh`` is executed during run time to startup and initialize the container. Here is what it's trying to accomplish:
+
+* Create Salt Config.
+* Set extra ENVs to NGINX.
+* Check if Database is ready.
+* Run Salt Scripts to establish the necessary set up for the docker image.
+* Fix permissions.
+* Start supervisor.
+* Showing the logs for supervisor, nginx and tethys.
+
+Run.sh also has these following optional arguments:
+
++---------------------------+------------------------------------------------------------------------------------------+
+| Argument                  | Description                                                                              |
++===========================+==========================================================================================+
+| --background              | run supervisord in background.                                                           |
++---------------------------+------------------------------------------------------------------------------------------+
+| --skip-perm               | skip fixing permissions step.                                                            |
++---------------------------+------------------------------------------------------------------------------------------+
+| --db-max-count            | number of attempt to connect to the database. Default is at 24.                          |
++---------------------------+------------------------------------------------------------------------------------------+
+| --test                    | only run salt scripts.                                                                   |
++---------------------------+------------------------------------------------------------------------------------------+
+
+***********
+Salt Script
+***********
+
+Tethys uses `Salt Script`_ to setup the app when the docker container runs. The file, named ``top.sls``, contains a list of state files to run. These files are ``pre_tethys.sls``, ``tethyscore.sls`` and ``post_app.sls``. You can override this file with your own ``top.sls`` file to insert a salt state file for your app. Here is an example of a ``top.sls`` file in a tethys app:
+
+::
+
+    base:
+      '*':
+        - pre_tethys
+        - tethyscore
+        - tethys_app
+        - post_app
+
+In this example, you can put logic needed to initialize your app in the ``tethys_app.sls`` file. The rest of the scripts are coming from tethys-core to help initialize the app setup up. Don't forget to add a line to the Dockerfile to add the new ``tethys_app.sls`` script to the ``/srv/salt`` directory:
+
+::
+
+    ADD tethys_app.sls /srv/salt/
```

Comment:
Add descriptions of what each salt file does with links to the files in the source code on GitHub.

Reply:
Done